src/components/Dropdown/DropdownInput.tsx

```diff
@@ -37,15 +37,15 @@ export function DropdownInput(props: {
 
   const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
     toggleDropdown(true);
+    onChange?.(e.target.value);
     updateFocusedItem(-1, e.target.value);
     setLastTypedOrSubmittedValue(e.target.value);
-    onChange?.(e.target.value);
   }, [onChange, setLastTypedOrSubmittedValue, toggleDropdown, updateFocusedItem]);
 
   const handleKeyDown = useCallback((e: KeyboardEvent<HTMLInputElement>) => {
     if (e.key === 'Enter' && (!submitCriteria || submitCriteria(focusedIndex))) {
```

Comment:
is this concept of `submitCriteria` needed anymore? if `alwaysSelectOption` is true, then we wouldn't need to check if it meets the criterion of having `focusedIndex >= 0` because that would always be the case, right?